### PR TITLE
NDI-87: Misc improvements to scripts

### PR DESCRIPTION
**Pull request by AkA84**
This PR contains several improvements + some clean up

### Before
Before, an option in a select2 dropdown would be selected with these two tsteps
```js
await page.openSelect2DropDown('#s2id_task'); // open the select2 with the given id
await page.clickSelect2NthOption(1); // select the option with the given index
```
This approach had two disadvantages:
1. It is cumbersome, always having to first open the dropdown before being able to select the option
2. It uses the index of the option, which is due to create problems in case the options will change positions (ie. the "Add activity" option will be moved from the 1st to the 3rd position in the dropdown

### After
The new approach uses a single method
```js
await page.clickSelect2Option('#s2id_task', 'Add activity');
```
The method params are the select2 selector (what was previously passed to `openSelect2DropDown`), and the option *label*

### Before
This is how a typical search action script (= a script that would perform a basic search, then select a contact, then select an action on it) would look like
```js
await engine.click('#_qf_Basic_refresh');
await engine.waitForNavigation();
await engine.click('#mark_x_3');
await engine.waitFor('#search-status .select2-container:not(.select2-container-disabled)');
await page.clickSelect2Option('#s2id_task', 'Add activity');
await engine.waitForNavigation();
```
These were the problems that are addressed in the PR
1. Lots of repetition. All the lines above the `clickSelect2Option` call are always the same for all the scripts.
2. The script performs a search with no filters, which then requires the DB to perform a slow query, which in turns slows down BackstopJS

### After
The repeated lines are now placed in a `search/actions/common.js` script, which can then be simply required in each search action script.

```js
module.exports = async (page, checkboxIds=['3']) => {
  await page.engine.type('#sort_name', 'Technology');
  await page.clickSelect2Option('#s2id_contact_type', 'Organization');
  await page.engine.click('#_qf_Basic_refresh');
  await page.engine.waitForNavigation();

  for (const id of checkboxIds) {
    await page.engine.click(`#mark_x_${id}`);
  }

  await page.engine.waitFor('#search-status .select2-container:not(.select2-container-disabled)');
};
```
The common script defines some query parameters to restrict the dataset returned by the query, thus improving the search performance. It also accepts multiple contact's checkbox ids in case a script needs to select multiple contacts before selecting an action

## Cleaned up `engine_scripts/` folder
The PR removes the `engine_scripts/chromy` and `engine_scripts/casper` folders as they are not necessary

### CODE REVIEW

File: backstop_data/engine_scripts/puppet/search/actions/common.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Performs a contact search, then selects the checkboxes with the given ids
+ *
+ * The search params are defined so that the smallest useful recordset is returned
+ * as a way to improve performance
+ *
+ * @param {CrmPage} page
+ * @param {Array} checkboxIds
+ */
+module.exports = async (page, checkboxIds=['3']) => {
```
Comment (deb1990) [inline backstop_data/engine_scripts/puppet/search/actions/common.js]: Why there is a default id `3`?

**Comment by deb1990** [comment]
@AkA84 as Backstop supports multiple engines like casper/chromy/puppet or something will come in future. Maybe it makes more sense to keep all the engine scripts in their respective folders(In this case puppet, as it was before)?

File: backstop_data/engine_scripts/puppet/search/actions/common.js (hunk 1 above)
Comment (AkA84) [reply backstop_data/engine_scripts/puppet/search/actions/common.js]: It's just the default contact's checkbox that gets selected by 99% of the search action scripts

**Comment by AkA84** [comment]
@deb1990

**Comment by AkA84** [comment]
@deb1990 your suggestion made sense, i rebased and pushed a new comment that only deletes chromy/ and casper/, but leaves puppet/ in place

**Pull request merged**